WalletWasabi/Stores/TransactionSqliteStorage.cs

```diff
@@ -0,0 +1,535 @@
+			return false;
+		}
+
+		tx = ReadRow(reader);
+		return true;
+	}
+
+	public IEnumerable<SmartTransaction> GetAll(CancellationToken cancellationToken)
+	{
+		using SqliteTransaction transaction = Connection.BeginTransaction();
+		using SqliteCommand command = Connection.CreateCommand();
+
+		// Note that a transaction can be in the mempool, so it does not have any specific block height assigned. However, we assign such 
+		// transactions Int32.MaxValue-1 height and as such mempool transactions would be returned first.
+		command.CommandText = $$"""
+			SELECT {{AllColumns}}
+			FROM "transaction"
+			ORDER BY block_height, block_index, first_seen
+			""";
+
+		using SqliteDataReader reader = command.ExecuteReader();
+
+		int i = 0;
+		while (reader.Read())
+		{
+			i++;
+
+			if (i % 100 == 0)
+			{
+				cancellationToken.ThrowIfCancellationRequested();
+			}
+
+			SmartTransaction filter = ReadRow(reader);
+			yield return filter;
+		}
+	}
+
+	public IEnumerable<uint256> GetAllTxids()
+	{
+		using SqliteTransaction transaction = Connection.BeginTransaction();
+		using SqliteCommand command = Connection.CreateCommand();
+
+		command.CommandText = $$"""
+			SELECT txid
+			FROM "transaction"
+			ORDER BY block_height, block_index, first_seen
+			""";
+
+		using SqliteDataReader reader = command.ExecuteReader();
+
+		while (reader.Read())
+		{
+			uint256 txid = new(reader.GetFieldValue<byte[]>(ordinal: 0));
+			yield return txid;
+		}
+	}
+
+	private SmartTransaction ReadRow(SqliteDataReader reader)
+	{
+		uint256 txid = new(reader.GetFieldValue<byte[]>(ordinal: 0));
+		int blockHeight = reader.GetInt32(ordinal: 1);
+		uint256? blockHash = reader.IsDBNull(ordinal: 2) ? null : new(reader.GetFieldValue<byte[]>(ordinal: 2));
+		int blockIndex = reader.GetInt32(ordinal: 3);
+		string labelsString = reader.GetString(ordinal: 4);
+		long firstSeenLong = reader.GetInt64(ordinal: 5);
+		bool isReplacement = reader.GetInt32(ordinal: 6) == 1;
+		bool isSpeedup = reader.GetInt32(ordinal: 7) == 1;
+		bool isCancellation = reader.GetInt32(ordinal: 8) == 1;
+		byte[] tx = reader.GetFieldValue<byte[]>(ordinal: 9);
+
+		Transaction transaction = Transaction.Load(tx, Network);
+
+		Height height = new(blockHeight);
+		LabelsArray labelsArray = new(labelsString);
+		DateTimeOffset firstSeen = DateTimeOffset.FromUnixTimeSeconds(firstSeenLong);
+
+		// TODO: Assert stx.txid against uint256 txid = new(reader.GetFieldValue<byte[]>(ordinal: 0));?
+		SmartTransaction stx = new SmartTransaction(transaction, height, blockHash, blockIndex, labelsArray, isReplacement, isSpeedup, isCancellation, firstSeen);
+
+		if (stx.GetHash() != txid)
+		{
+			throw new InvalidOperationException();
+		}
+
+		return stx;
+	}
+
+	/// <returns>Returns <c>true</c> if there are no records in the transactions table.</returns>
+	public bool IsEmpty()
+	{
+		using SqliteCommand command = Connection.CreateCommand();
+		command.CommandText = """SELECT EXISTS (SELECT 1 FROM "transaction")""";
```

Comment:
Don't know maybe some .Net details, but is the subselect really necessary here and just `SELECT 1 FROM "transaction"` doesn't give the same result? For SQLite True / False is just aliases of 1 / 0. Also `LIMIT 1` should be added at the end.